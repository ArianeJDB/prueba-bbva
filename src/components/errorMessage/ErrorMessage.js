import React from 'react';
import './errorMessage.css';

const ErrorMessage = ({ errorMsg }) => {
    return (
            <div className="error">{errorMsg}</div>
    )
}

export default ErrorMessage;