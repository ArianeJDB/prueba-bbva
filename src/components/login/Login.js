import React, { useState } from 'react';
import { Link } from 'react-router-dom';


const Login = () => {
    return (
        <Link to={`/counter`}> <div>login</div></Link>
       
    )
}

export default Login;