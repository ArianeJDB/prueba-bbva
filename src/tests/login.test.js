import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import Login from '../components/login/Login'
import ErrorMessage from '../components/errorMessage/ErrorMessage'

describe('Login component should', () => {
    let wrapper;
    test('should render correctly', () => {
        wrapper = shallow(<Login />)

        expect(wrapper).toMatchSnapshot()
    })

    test('should render ErrorMsg with errorMsg as a prop', () => {
        const errorMsg = 'Email is required'
        wrapper = shallow(<ErrorMessage errorMsg={errorMsg} />)

        expect(wrapper).toMatchSnapshot()
    })

})
