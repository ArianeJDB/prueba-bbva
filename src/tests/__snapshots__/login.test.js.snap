// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component should should render ErrorMsg with errorMsg as a prop 1`] = `
<Fragment>
  <div
    className="error"
  >
    Email is required
  </div>
</Fragment>
`;

exports[`Login component should should render correctly 1`] = `
<div
  className="login_container"
>
  <i
    className="fa fa-lock"
  />
  <form
    onSubmit={[Function]}
  >
    <label
      className="hidden"
      htmlFor="email"
    >
      Email
    </label>
    <input
      name="email"
      placeholder="Email"
      type="text"
    />
    <label
      className="hidden"
      htmlFor="password"
    >
      Contrase√±a
    </label>
    <input
      name="password"
      placeholder="Password"
      type="password"
    />
    <button>
      Log in
    </button>
  </form>
</div>
`;
